apps/test/unit/javalab/JavalabEditorTest.js

```diff
@@ -56,338 +57,419 @@ describe('Java Lab Editor Test', () => {
-      expect(javalabEditor.state.orderedTabKeys).to.deep.equal([
-        'file-0',
-        'file-1'
-      ]);
-      expect(javalabEditor.state.fileMetadata).to.deep.equal({
-        'file-0': 'Class1.java',
-        'file-1': 'Class2.java'
+
+      it('can delete the only file in the editor', () => {
+        const editor = createWrapper();
+        const javalabEditor = editor.find('JavalabEditor').instance();
+
+        javalabEditor.setState({
+          showMenu: false,
+          contextTarget: null,
+          openDialog: 'createFile',
+          orderedTabKeys: ['file-0'],
+          lastTabKeyIndex: 1,
+          fileMetadata: {
+            'file-0': 'Class1.java'
+          },
+          activeTabKey: 'file-0',
+          fileToDelete: 'file-0'
+        });
+        javalabEditor.onDeleteFile();
+        expect(store.getState().javalab.sources['Class1.java']).to.be.undefined;
+        expect(javalabEditor.state.openDialog).to.be.null;
+        expect(javalabEditor.state.activeTabKey).to.be.null;
+        expect(javalabEditor.state.orderedTabKeys).to.deep.equal([]);
+        expect(javalabEditor.state.fileMetadata).to.deep.equal({});
       });
     });
-  });
 
-  describe('Delete File', () => {
-    it('updates state and sources on delete save', () => {
-      const editor = createWrapper();
-      const javalabEditor = editor.find('JavalabEditor').instance();
-
-      javalabEditor.setState({
-        showMenu: false,
-        contextTarget: null,
-        openDialog: 'createFile',
-        orderedTabKeys: ['file-0', 'file-1'],
-        lastTabKeyIndex: 1,
-        fileMetadata: {
-          'file-0': 'Class1.java',
-          'file-1': 'Class2.java'
-        },
-        activeTabKey: 'file-0',
-        fileToDelete: 'file-0'
+    describe('Read Only Mode', () => {
+      it('is editable', () => {
+        const editor = createWrapper();
+        const javalabCodeMirrors = editor.find('JavalabEditor').instance()
+          .editors;
+        const firstEditor = Object.values(javalabCodeMirrors)[0];
+
+        expect(firstEditor.state.facet(EditorView.editable)).to.be.true;
       });
-      javalabEditor.onDeleteFile();
-      expect(store.getState().javalab.sources['Class1.java']).to.be.undefined;
-      expect(javalabEditor.state.openDialog).to.be.null;
-      expect(javalabEditor.state.orderedTabKeys).to.deep.equal(['file-1']);
-      expect(javalabEditor.state.fileMetadata).to.deep.equal({
-        'file-1': 'Class2.java'
+
+      it('header buttons are enabled', () => {
+        const editor = createWrapper();
+        editorHeaderButtonIdentifiers.forEach(headerButtonId => {
+          const isButtonDisabled = editor
+            .find(headerButtonId)
+            .first()
+            .props().isDisabled;
+
+          expect(isButtonDisabled).to.be.false;
+        });
       });
     });
+  });
 
-    it('can delete the only file in the editor', () => {
+  describe('View Only Mode', () => {
```

Comment:
I had to move existing tests into their own `describe` block b/c I wanted to set a page constant for these new tests, which can't be reset. Most of the changes to this file are just from that move -- the tests in the 'View Only Mode' block are new.